src/components/Delegate/StakingOpportunities.tsx

```diff
@@ -11,36 +11,37 @@ import {
 } from '@chakra-ui/react'
 import { CAIP19 } from '@shapeshiftoss/caip'
 import { AprTag } from 'plugins/cosmos/components/AprTag/AprTag'
-import { MouseEvent } from 'react'
-import { useMemo } from 'react'
+import { MouseEvent, useEffect, useMemo } from 'react'
 import { NavLink } from 'react-router-dom'
-import { Column } from 'react-table'
+import { Column, Row } from 'react-table'
 import { Amount } from 'components/Amount/Amount'
 import { AssetIcon } from 'components/AssetIcon'
 import { Card } from 'components/Card/Card'
 import { ReactTable } from 'components/ReactTable/ReactTable'
 import { RawText, Text } from 'components/Text'
 import { useModal } from 'hooks/useModal/useModal'
-import { BigNumber, bnOrZero } from 'lib/bignumber/bignumber'
-import { selectAssetByCAIP19 } from 'state/slices/selectors'
-import { useAppSelector } from 'state/store'
-
-type StakingOpportunity = {
-  id: number
-  moniker: string
-  apr: BigNumber
-  cryptoAmount?: BigNumber
-  rewards: Rewards
-}
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import {
+  selectAssetByCAIP19,
+  selectMarketDataById,
+  selectPubkeyishByChainId
+} from 'state/slices/selectors'
+import {
+  ActiveStakingOpportunity,
+  ASSET_ID_TO_DENOM,
```

Comment:
As we've agreed with @elmutt, denom references should go away in web

Reply:
fixed ! removed all denom references and compared with assetId instead